xetusoss-archiva/CHANGELOG.md

```diff
@@ -2,6 +2,13 @@
 ---
 Notable changes in the xetusoss-archiva chart
 
+### v0.1.5
+
+* Added better support for `v2` images after the [`archiva:v2.2.4-1`](https://github.com/xetus-oss/docker-archiva#v224-1) changes. The new image version sets the `XX:+UseContainerSupport` flag so `cgroups` are respected.
+  * Renamed `resourceConf.jvmMaxMemory` to `resourceConf.memory` and adjusted the default to `512`. Backwards compatibility with the `resourceConf.jvmMaxMemory` will be supported until chart version `v0.2.0`.
+  * The `JVM_MAX_MEMORY` environment variable is no longer set in Archiva container since support for this has been dropped. __WARNING:__ If an earlier version tag of the `xetusoss/archiva` image is manually specified, the `JVM_MAX_MEMORY` variable will not be set and the image's default settings will be used.
```

Comment:
This sounds like this version of the chart provides no way to configure the max memory consumed by the JVM in pre-v2.2.4-1 containers and are then stuck with the image's default while still being able to modify the resource limits and requests. If that's true, then users could end up in a position where the JVM can consistently consume more than the configured memory limit without being able to reconfigure the JVM memory limit, leading to Kubernetes repeatedly OOM killing the pod.

If that's right I think one of two things should happen:

1. this should be a minor release (`v0.2.0`) and just remove attempts at backwards compatibility;
2. the `JVM_MAX_MEMORY` variable should be passed to the pod only if the `resourceConf.jvmMaxMemory` value is supplied.

Reply:
I can  see why you'd say that, and I considered it, but there really is no reason to preserve backwards compatibility. 

>  If that's true, then users could end up in a position where the JVM can consistently consume more than the configured memory limit without being able to reconfigure the JVM memory limit, leading to Kubernetes repeatedly OOM killing the pod.

That's actually the case in all previous versions, even when the `JVM_MAX_MEMORY` was controlling the HEAP size! Due to how the JVM was allocating memory, no matter what you set the `JVM_MAX_MEMORY` to, the pod would eventually over-allocate memory and be killed.

> The JVM_MAX_MEMORY variable should be passed to the pod only if the resourceConf.jvmMaxMemory value is supplied.

I could do this, but would definitely lead to a false sense of security. We've found that setting this value _does not_ actually constraint the JVM's max memory.

Reply:
This makes more sense then -- "backwards compatible" here really refers to compatibility with the `resourceConf.jvmMaxMemory` configuration value, not to previous behavior (which was always broken).

Thanks for explaining that!